robottelo/host_info.py

```diff
@@ -64,7 +64,7 @@ def _extract_sat_version(ssh_cmd):
     ssh_result = ssh.command(ssh_cmd)
     if ssh_result.stdout:
         version_description = ssh_result.stdout[0]
-        version_re = r'[^\d]*(?P<version>\d(\.\d){1})'
+        version_re = r'[^\d]*(?P<version>\d(\.\d){2})'
```

Comment:
That code is executed only if `SATELLITE_VERSION` environmental variable is not set, and AFAIK it will always be set on current Jenkins pipeline. What's worse - current Jenkins pipeline assume `SATELLITE_VERSION` to be `major.minor`, without z-stream number included.

This effectively causes difference in version content depending on environment robottelo is run on. version will usually include z-stream number locally, but will never include it on Jenkins. I'm certain this difference is going to bite us one day.

But I think you don't really need this, as change in `pytest_plugins/rerun_rp/rerun_rp.py` already discards z-stream number?

Reply:
@mirekdlugosz great point, and a pattern that I believe we are eliminating in the new CI pipelines. @lpramuk Thoughts on this behavior in the new CI pipelines - The current open MR from @jawatts is not setting SATELLITE_VERSION env var.

Overall I think the version pulled from the target satellite should override the envvar, because that's the _real_ version that we're running against. SATELLITE_VERSION should only be used if there's no target satellite, or because it can't be reached.

Reply:
@mirekdlugosz If the Jenkins pipeline assumes the Satellite Version as major.minor then should either be fixed by considering the `micro` appending to it or Jenkins chops out micro if its being used from this function. And I am sure the jenkins pipeline isnt fetching the version from this function. Its the static satellite version there.

As far as this PR concerns, I checked the usage of this function all over robottelo and it doesn't seems to bite anywhere in robottelo. Also, the rerun_rp isnt discarding the `z-stream`, I am removing `version.epoch` as it always returns `0` for zstream and ystream satellite.

Reply:
@mshriver Good point out there. We should use actual extracted version from Satellite. Also, Whats the way @jawatts / @lpramuk is using the to extract sat version from sat ?

Reply:
Overall, This PR helps us in unblocking zStream rerun and hence we have to weigh this PR with Jenkins changes and get it in.

Reply:
@jyejare right now the pipelines aren't pulling satellite version from running instances, has no need to. Its receiving specific UMB messages with x.y.z versions to trigger jobs.  

That being said, I think we should start to be more specific with the version read from running Satellites, and adjust pipelines and run environments as needed.

The pipelines are build for X.Y versions, so there are going to be points where we'll have to deal with some version processing. @jyejare I trust you'll be accountable to dealing with those changes, and updating the team on venv version setting.

Reply:
@mshriver As per discussion on GMeet which in short describes that, 'We will be shortly moving to new Pipelines where we dont sets SATELLITE_VERSION and the plan is to read the satellite version as 6.y.z. And you have ACKed that this PR is on track of that.'.

As to prove this PR doesn't breaks the existing pipeline, I have ran Standalone Automation Job (#2097 with Tier1 tests)  and I dint see any unusual failure caused due to this PR. 

Also, I have removed the Satellite 6.1 way of retrieving Satellite Version from sat by looking into version.rb file as its no longer required in recent releases (rpm -q satellite is enough). Also the version.rb file is removed from recent releases and I am not aware any other file which can be replaced here.

The Travis in now fixed.